WoWsShipBuilder.Common/Infrastructure/DataTransfer/AppPagesBrowser.cs

```diff
@@ -0,0 +1,176 @@
+﻿using Microsoft.AspNetCore.Components;
+using MudBlazor;
+using WoWsShipBuilder.Infrastructure.Metrics;
+
+namespace WoWsShipBuilder.Infrastructure.DataTransfer;
+
+public class AppPagesBrowser
+{
+    private const int TransferLimit = 250;
+
+    private const string ShipStatsBaseUrl = "ship?shipIndexes=";
+    private const string BallisticChartsBaseUrl = "charts?shipIndex=";
+    private const string AccelerationChartsBaseUrl = "acceleration-charts?shipIndexes=";
+    private const string ShipComparisonBaseUrl = "comparison?p=";
+
+    private readonly SessionStateCache sessionStateCache;
+    private readonly NavigationManager navManager;
+    private readonly MetricsService metricsService;
+
+    public AppPagesBrowser(SessionStateCache sessionStateCache, NavigationManager navManager, MetricsService metricsService)
+    {
+        this.sessionStateCache = sessionStateCache;
+        this.navManager = navManager;
+        this.metricsService = metricsService;
+    }
+
+    public enum AppPage
+    {
+        ShipStats,
+        BallisticCharts,
+        AccelerationCharts,
+        ShipComparison,
+    }
+
+    private void GoToPage(string destinationPageBaseUrl, List<ShipBuildContainer> containerList, string? shellIndex, ISnackbar? snackbar, AppPage? leavingPage, string metricLabel)
+    {
+        int selectionCount = containerList.Count;
+        switch (selectionCount)
+        {
+            case > TransferLimit:
+                snackbar?.Add($"Too many ships. Eligible ships: {selectionCount}. Max limit is {TransferLimit}.", Severity.Error);
+                return;
+            case 0:
+                snackbar?.Add($"No ships available. Eligible ships: {selectionCount}.", Severity.Error);
+                return;
+            default:
+                this.LogMetrics(leavingPage, destinationPageBaseUrl, metricLabel);
+                this.sessionStateCache.SetBuildTransferContainers(containerList);
+                this.navManager.NavigateTo(destinationPageBaseUrl + string.Join(',', containerList.Select(x => x.Ship.Index)) + (shellIndex is not null ? $"&shellIndex={shellIndex}" : string.Empty));
```

Comment:
too many checks for computing urls. Should probably be separated into different cases for each page. This is just confusing to read